### DISCUSSION TITLE
Extraction Automation

### DISCUSSION THREAD

**Discussion by ericchernuka** [lingui/js-lingui#1942] [Q&A]
Just wondering how everyone is doing the extraction process. I saw the [git hook](https://lingui.dev/ref/cli) here which is totally valid, but does end up causing a lot of commit noise with the PO file constantly updating the line numbers. I noticed that rebasing when this occurs becomes a bit nightmarish. 

So I was just wondering if there are other techniques being used in other apps? CI workflow?

One other option I had considered was a CI step that ran extract and if it resulted in uncommitted files it would either auto commit them or throw an error indicating that the step needed to be completed.

Anyways, love to get some ideas from others. Thanks!

**Comment by timofei-iatsenko** [comment]
We use extraction to a template:

// package.json
```js
    "build": "yarn extract-messages && next build",
    "extract-messages": "lingui extract-template",
```

This way your build is always up-to-date (no ID's instead of messages). 

For translation process we use TMS (translation management system) and have a separate task for that in CI called "extract messages" which we call when we ready to translate 

The tasks is a simple "yarn extract-messages && crowdin upload"

You don't need to update your catalogs on every commit, this will cause a lot of conflicts between branches. Simply write your code and treat extraction artifacts as derivative of your code (like a produced .map from your .ts file)

**Reply by ericchernuka** [reply]
Thanks @thekip! In your opinion, if we just need English for the foreseeable feature should we just avoid the extraction process steps entirely until we need to add another language?

**Reply by timofei-iatsenko** [reply]
Not at all. If you omit your extraction process, you will get id's instead of messages, since lingui drops original messages in production build. So your builds/bundles doesn't contain any "default" language baked in. You need to add an extraction 
step before build as I showed above, so every time you build an application you have up to day template catalog.

**Comment by quantum-incongruity** [comment]
We couldn't add a flag "--no-line-number" to the extract ?

**Reply by timofei-iatsenko** [reply]
Nope, because this is formatter related settings. Lingui cli knows nothing about how the catalog would be serialized.

**Reply by quantum-incongruity** [reply]
Okay thanks https://lingui.dev/ref/catalog-formats

```js
import { defineConfig } from "@lingui/cli";
import { formatter } from "@lingui/format-po";

export default defineConfig({
  // [...]
  format: formatter({ lineNumbers: false }),
});
```
